ui/src/main/java/com/hyperwallet/android/ui/repository/TransferMethodConfigurationRepositoryImpl.java

```diff
@@ -136,38 +137,53 @@ public Handler getHandler() {
 
     @Override
     public synchronized void getKeys(@NonNull final LoadKeysCallback loadKeysCallback) {
-        if (mTransferMethodConfigurationKeyResult == null) {
+        if (mTransferMethodConfigurationKey == null) {
             getTransferMethodConfigurationKeyResult(loadKeysCallback);
         } else {
-            loadKeysCallback.onKeysLoaded(mTransferMethodConfigurationKeyResult);
+            loadKeysCallback.onKeysLoaded(mTransferMethodConfigurationKey);
         }
     }
 
     @Override
     public synchronized void getFields(@NonNull final String country, @NonNull final String currency,
             @NonNull final String transferMethodType,
             @NonNull final LoadFieldsCallback loadFieldsCallback) {
+
         FieldMapKey fieldMapKey = new FieldMapKey(country, currency, transferMethodType);
-        HyperwalletTransferMethodConfigurationFieldResult transferMethodConfigurationFieldResult = mFieldMap.get(
-                fieldMapKey);
+        HyperwalletTransferMethodConfigurationField transferMethodConfigurationField = mFieldMap.get(fieldMapKey);
         // if there is no value for country-currency-type combination,
         // it means api call was never made or this combination or it was refreshed
-        if (transferMethodConfigurationFieldResult == null) {
+        if (transferMethodConfigurationField == null) {
             getTransferMethodConfigurationFieldResult(country, currency, transferMethodType, loadFieldsCallback);
         } else {
-            loadFieldsCallback.onFieldsLoaded(transferMethodConfigurationFieldResult);
+            loadFieldsCallback.onFieldsLoaded(transferMethodConfigurationField,
+                    getProcessingTime(country, currency, transferMethodType));
         }
     }
 
     @Override
     public void refreshKeys() {
-        mTransferMethodConfigurationKeyResult = null;
+        mTransferMethodConfigurationKey = null;
     }
 
     @Override
     public void refreshFields() {
         mFieldMap.clear();
     }
+
+    @Nullable
+    private String getProcessingTime(String country, String currency, String transferMethodType) {
```

Comment:
@peter-joseph What about tests for getProcessingTime? Maybe we can remove private (access modifier)

Reply:
That's a temporary method, I added todo in other PR